### ISSUE TITLE
Support Rails's `attribute` in Cancancan's `permitted_attributes`

### ISSUE THREAD

**Issue by kalsan** [CanCanCommunity/cancancan#838]
TLDR: Cancancan does not "see" non-database backed attributes, so they are missing in `permitted_attributes`.

### Steps to reproduce

Rails 7.1.2 application with a basic user model with the database fields `email` and `password_digest`, with the important parts from the model being:

```ruby
class User < ApplicationRecord

has_secure_password
```

With the god-ability:

```
class Ability
  include CanCan::Ability

  attr_reader :user

  def initialize(user)
    return if user.blank?

    can :manage, :all
  end
end
```

Now run: `Ability.new(User.first).permitted_attributes(:edit, User.first)`

### Expected behavior

```ruby
[:email, :password_digest, :password, :password_confirmation]
```

### Actual behavior

```ruby
[:email, :password_digest]
```

-> Cancancan misses out on the password attributes. The same problem can be reproduced by using Rail's `attribute` method, i.e.

```ruby
class MyModel < ApplicationRecord
  attribute :foo # Cancancan will miss this
end
```

### System configuration
**Rails version**: 7.1.2

**Ruby version**: 3.2.2

**CanCanCan version** 3.4.0

I guess this also interferes with other Gems such as `active_type` or any custom accessor logic using virtual fields using `attribute`.

I hope the examples above are useful to you. Let me know if I can help you enlighten this further.

Best,
Kalsan
